### PR TITLE
Use DynFilesystem trait for FS operations

### PR DESCRIPTION
**Pull request by robin-nitrokey**
This reduces code complexity and binary size.

### CODE REVIEW

File: src/store/filestore.rs [hunk 1]
```diff
@@ -197,21 +196,22 @@ impl<S: Store> ClientFilestore<S> {
             })
             .ok())
     }
-    fn read_dir_next_impl<F: LfsStorage + 'static>(
+    fn read_dir_next_impl(
         &mut self,
         state: ReadDirState,
-        fs: &'static Fs<F>,
     ) -> Result<Option<(DirEntry, ReadDirState)>> {
+        // TODO: check what happens (or used to happen) for FS != IFS
```
Comment (sosthene-nitrokey) [inline src/store/filestore.rs:203]: Why?

File: src/store/filestore.rs [hunk 2]
```diff
@@ -280,7 +280,8 @@ impl<S: Store> ClientFilestore<S> {
                             real_dir: dir.clone(),
                             last: i,
                             location,
-                            user_attribute,
+                            // TODO: check if we can avoid that clone
```
Comment (sosthene-nitrokey) [inline src/store/filestore.rs:283]: The `Dyn` impl could actually take an `FnOnce`, turn it into an `FnMut` that panics if invoked more than once.

**Review by sosthene-nitrokey — APPROVED** [review]

File: src/store/filestore.rs (hunk 1 above)
Comment (robin-nitrokey) [reply src/store/filestore.rs:203]: I think I saw a place where this was always called with the IFS even for other locations, but I can’t find it any more.  Maybe I accidentally looked at the `locate_file` implementation.

Comment (robin-nitrokey) [reply src/store/filestore.rs]: Removed the comment.

File: src/store/filestore.rs (hunk 2 above)
Comment (robin-nitrokey) [reply src/store/filestore.rs:282]: I think we should also be able to just move that part of the code out of the closure.

Comment (robin-nitrokey) [reply src/store/filestore.rs]: Yeah, that works and also gets rid of the dir clone.

**Pull request merged**